sdcm/nemesis.py

```diff
@@ -2141,6 +2142,28 @@ def choose_partitions_for_delete(self, partitions_amount, ks_cf, with_clustering
         self.log.debug(f'Partitions for delete: {partitions_for_delete}')
         return partitions_for_delete
 
+    def get_random_timestamp_from_partition(self, ks_cf, pkey, partition_percentage=0.25) -> int:
+        with self.cluster.cql_connection_patient(node=self.target_node) as session:
+            partition_size = session.execute(
+                f"select count(ck) from {ks_cf} where pk = {pkey}").one().system_count_ck
+            fetch_limit = max(math.ceil(partition_size * partition_percentage), 11)
+            self.log.debug(
+                "[%s_using_timestamp] Partition size: %s, fetching up to %s",
+                self.get_disrupt_name(),
+                partition_size,
+                fetch_limit
+            )
+            partition = session.execute(f"select pk, ck from {ks_cf} where pk = {pkey} limit {fetch_limit}").all()
+            delete_mark = partition[-1].ck
+            timestamp = session.execute(
+                f"select writetime(v) from {ks_cf} where pk = {pkey} and ck = {delete_mark}").one().writetime_v
+            if not timestamp:
+                message = f"Unable to get writetime for row (pk = {pkey}, ck = {delete_mark})"
+                self.log.error(message)
+                raise Exception(message)
```

Comment:
Maybe better to create specific exception class and use here?
Not sure it is good idea to use the `Exception` directly.

Reply:
Will do